### ISSUE TITLE
RTMP and encoder 

### ISSUE THREAD

**Issue by ClearStreamDevTeam** [WWBN/AVideo#9705]
Hello. We are having issues with our RTMP and encoder. Can you please assist. Everything else is working nice!

**Comment by milehighkaraoke** [comment]
What is the exact problem

**Comment by milehighkaraoke** [comment]
Clear logs, run a live broadcast, get logs, post logs.

**Comment by milehighkaraoke** [comment]
Update both encoder and site before you pull logs

**Comment by ClearStreamDevTeam** [comment]
Ok. I am doing that today. I also have an IP ban on the encoder. Not sure why that is when My server is doing the encoding

**Comment by ClearStreamDevTeam** [comment]
Hi milehighkaraoke,

Here is the screenshot of my server config on the script. mbstring is installed, also the ports on nginx is closed. I configured the nginx server but there has to be something I am missing
![server screenshot](https://github.com/user-attachments/assets/2d07bc86-3178-46ca-9638-6314582f2e22)

**Comment by milehighkaraoke** [comment]
Yes, you have to open those ports on your server and port forward them from your router. You have to install mbstring and get your schedule installed in the crontab.

**Comment by milehighkaraoke** [comment]
Make sure all plug ins are updated.. even those you are not using.

**Comment by ClearStreamDevTeam** [comment]
Hi milehighkaraoke,

I also find my CPU usage strange. I have 2 xeon processors running at a total of 24 cores. 64GB RAM and my CPU usage is at 100% Please look at the screenshot
![avideo_100_cpu](https://github.com/user-attachments/assets/75ebc49a-7729-4178-9528-dc92a8aeaaa7)

**Comment by ClearStreamDevTeam** [comment]
right now I am trying to get nginx working with reverse proxy. there seems to be a conflict. I had to re-install nginx to a newer version.

**Comment by ClearStreamDevTeam** [comment]
here is the screenshot of the YPT socket plugin. I highlighted the icon that keeps going around in circles 
![Screenshot 2024-12-17 113228](https://github.com/user-attachments/assets/143b74c3-f8a3-4891-8d28-2c408480235e)

**Comment by milehighkaraoke** [comment]
so, have you opened the required ports on your server? It looks as though a firewall is blocking your traffic
Also, your schedular has to be running for all this stuff to work

**Comment by milehighkaraoke** [comment]
High CPU is probably attributed to site, Nginx, Apache and schedular trying to communicate

**Comment by ClearStreamDevTeam** [comment]
Yes port 8080, and 8443 are open. port 1935 has an error

The nginx plugin is not working; there may be problems with your existing configuration.
The error was: MisconfigurationError('Error while running nginx -c /usr/local/nginx/conf/nginx.conf -t.\n\nnginx: [emerg] unknown directive "rtmp             " in /usr/local/nginx/conf/nginx.conf:7\nnginx: configuration file /usr/local/nginx/conf/nginx.conf test failed\n')

**Comment by ClearStreamDevTeam** [comment]
I am running nginx 1.26.2

**Comment by ClearStreamDevTeam** [comment]
ok so all ports are open. my live shows the thumbnail from OBS but nothing streams or records after live stream. it's getting there. do you have any suggestions

**Comment by milehighkaraoke** [comment]
Put your schedular in your crontab. 
Directions are under system health

**Comment by ClearStreamDevTeam** [comment]
Ok so good news. most ports are open expect for 2053. mbring is enabled. There are 2 issues I would like to get your thoughts on. 

1. When I stream from OBS, my thumbnail shows on the script under live but I get no audio/video. the stats in the RTMP show its recording from the root HLS/live folder however it doesn't broadcast on the script or encode it as a recording. 
2. my CPU still shows 100% I have 24 cores Xeon Processing Power at 2.93GHZ per processor. I have 2 physical with 64GB ECC RAM. my server is a Dell R710. Please see the screenshot

Do you have any thoughts about this. It is almost there . Thank you in advance, you have been amazing helping us out with this!
![Srerver Info](https://github.com/user-attachments/assets/c8e5d22b-fc36-49ef-8bda-afdfdbdce9b7)
![CPU](https://github.com/user-attachments/assets/70a82fbc-20d4-4cc8-a87b-d47761969142)

**Comment by ClearStreamDevTeam** [comment]
YPT plugin working now. Its just the RTMP Videos not streaming and high CPU at this point :)

**Comment by milehighkaraoke** [comment]
ok, what do you get when you type localhost in the web browser?

**Comment by milehighkaraoke** [comment]
I am gonna pass this off to the main man. 
@DanielnetoDotCom I got him this far but am now drawing a blank. Are you able to give this a quick look ver.
Thank you

**Comment by ClearStreamDevTeam** [comment]
Thank you for your help! its 99% there :) here is the streamer
[ffmpeg_restreamer_1_2024-12-18-11-04-17_1_4_prostreams.live.log](https://github.com/user-attachments/files/18199617/ffmpeg_restreamer_1_2024-12-18-11-04-17_1_4_prostreams.live.log)
 log

**Comment by ClearStreamDevTeam** [comment]
here is my avidoe2.conf file 
[avideo2.conf.txt](https://github.com/user-attachments/files/18199823/avideo2.conf.txt)

**Comment by ClearStreamDevTeam** [comment]
@DanielnetoDotCom ,

@milehighkaraoke was kind enough to help us get our server fixed. We were successful with YPT WebSocket, and Nginx, however our avideo2.conf is eating up all CPU resources. The TRMP seems to be working to a point but not streaming on the script and not encoding after the live has ended. Can you please shed some light as we are about 99% there :) By the way Awesome platform!! so many features to utilize! You and your team did an amazing job on this platform script!

**Comment by ClearStreamDevTeam** [comment]
i just noticed nicehas in the config. This seems odd

**Comment by ClearStreamDevTeam** [comment]
Hello @milehighkaraoke and @DanielnetoDotCom !

I am happy to report that we have fixed the CPU Usage! the only issue is RTMP streaming and encoding live. Our Thumbnail shows up on the script that we push out to the server however the videos are not streaming or recording/coding after the live stream has ended. Everything else addressed in this ticket is resolved :) any thoughts on how to get the streamer and recording working

**Comment by milehighkaraoke** [comment]
Have you done a git pull on both your site and encoder?

**Comment by ClearStreamDevTeam** [comment]
Yes. and upgraded to 14.4

**Comment by milehighkaraoke** [comment]
What do you get when you copy and paste your on publish address into your servers address bar?

**Comment by ClearStreamDevTeam** [comment]
unsecure. but i deleted the encoder and re-insalling and stuck on processing. it would not install as of yet

**Comment by ClearStreamDevTeam** [comment]
Encoder installed with success and able to encode videos. The RTMP seems to be working and showing in the logs that my stream is live. In the script it even shows my thumbnail so I know its reporting. When I click on live nothing shows up and doesn't save the stream

**Comment by ClearStreamDevTeam** [comment]
We found this

2024/12/20 02:43:01 [info] 88035#0: *8194 client closed connection while SSL handshaking, client: 127.0.0.1, server: 0.0.0.0:8443
2024/12/20 02:43:01 [info] 88035#0: *8195 SSL_do_handshake() failed (SSL: error:0A000418:SSL routines::tlsv1 alert unknown ca:SSL alert number 48) while SSL handshaking, client: 127.0.0.1, server: 0.0.0.0:8443

**Comment by ClearStreamDevTeam** [comment]
Ok 99.9% there. live stream works but when the stream ends, it doesn't encode or save to videos in the script. do you have any thoughts

**Comment by milehighkaraoke** [comment]
send live to encoder plug-in?

**Comment by ClearStreamDevTeam** [comment]
Yes, I got the server to stream however if i want to start a new stream I have to reset the stream key to start a new stream. When I do that the title has (1) at the end.

**Comment by ClearStreamDevTeam** [comment]
here is an example test stream

https://prostreams.live/live/0/Sysadmin/1

**Comment by ClearStreamDevTeam** [comment]
here are streams from 2 accounts the platform has

https://prostreams.live/live/0/Sysadmin/1

https://prostreams.live/live/0/TrueCrimeWithGarrenTee/1

The streams are only available with those links and not pushed to the script under videos. Plus the key needs to be changed manually every time the users start a new stream. we are getting there :) do you have any thoughts

**Comment by ClearStreamDevTeam** [comment]
I may have to update my sendtoencoder. I tried logging into my account to get the updated plugin but my password or username is wrong. I emailed @DanielnetoDotCom but haven't heard back yet. I am sure he is busy working on tickets.

**Comment by ClearStreamDevTeam** [comment]
ok so I got the live stream to encode and save but now can't view it live, do you have any thoughts

**Comment by ClearStreamDevTeam** [comment]
Update. We now can view the stream live and encodes.  Is there a way to get rid of the (1) right after the live title and have the stream save without the date and time format.

**Comment by ClearStreamDevTeam** [comment]
Hello! Issues are now resolved. Thank you for your support!

**Comment by baoang** [comment]
About the avideo.conf that caused high cpu consumption, it's a virus, Linux/CVE-2021-4034.B!MTB.

kill it in htop and there are endless coming up. Stop the web server and delete the file, and it will generate similar files again.

I finally zipped the whole dir and downloaded it to my pc desktop and the anti-virus software found the contamination.

v14.4 if I am not mistaken. Finally I removed all the files except the video dir, and git pulled. Over two weeks' time, cpu played well.
